CMakeLists.txt

```diff
@@ -179,6 +188,10 @@ target_link_libraries (test_validator ${LIBRARIES_TO_LINK})
 enable_testing ()
 add_test (NAME ValidatorTests COMMAND test_validator)
 
+add_executable (test_assembly_checker test/main_test.cpp ${ASSEMBLY_CHECKER_TESTS})
+target_link_libraries (test_assembly_checker ${LIBRARIES_TO_LINK})
+enable_testing ()
+add_test (NAME AssemblyCheckerTests COMMAND test_assembly_checker)
```

Comment:
Now that we have 3 tools we should agree how to structure the test binaries. `ALL_TESTS` includes validator and debugulator tests, but the assembly checker ones are in a completely separate set. I would say that "ALL" should mean "ALL", but it is also true that we should make the granularity more clear and try to remove redundancies.

Any particular preference @jmmut ?

Reply:
I'm used to just run the `test_validator` binary which runs all tests. I'm not sure having more binaries adds value, given that we could use the tag system better to select a subset of tests.

Reply:
Good point about the tag system. Creating a single binary for all the tests should reduce the build time. @ManeshNarayan can you try replacing the current test command as suggested by @jmmut ? At least to split the validator and assembly tests in 2 runs for the sake of traceability.

Reply:
+1 at using ALL_TESTS. About the split, I meant using the tags to execute first only the validator tests, then the assembly tests for improved granularity. But it is not critical so I'm fine with running all of them in a single run.